docfx_project/articles/bootstrapping-trust/part-01.encryption-everywhere.md

```diff
@@ -0,0 +1,207 @@
+introduces some complex behaviors. Setup and management are not always
+as easy as one would hope.
+
+
+#### Setting It Up
+
+So we have decided that public-key cryptography is the answer. What does
+that mean? What do I have to do? Let us explore what would need to be
+done by a human or a piece of software automating this process. Don't
+worry if you don't get all of this; the gist is all you need for now.
+Later parts will expand upon this terminology. In fact, after reading
+parts two, three, and four, consider revisiting this part.
+
+Consider the following diagrams of a "mesh" distributed system. This
+system has multiple pieces of software, all connecting amongst
+themselves. Consider what it means to accomplish this using public-key
+cryptography.
+
+![Image of a simple mesh](./images/mesh.png)
+
+In the diagram above, each system needs:
+
+- a key pair for client & server connections
+- to have the public keys of everyone it is connecting to
+
+So what do we need to do? Drop into a CLI and start generating keys on
+each machine. Do that by using these commands:
+
+```
+openssl ecparam -name secp256k1-genkey -param_enc explicit -out private-key.pem
+```
+
+```
+openssl req -new -x509 -key private-key.pem -out server.pem -days 360
+```
+
+Voila - you now have a self-signed certificate that no one trusts. You
+can repeat this for every piece of software in your mesh network, as it
+initially will be deployed. Preferably, you log into each machine and
+generate the private key there. Moving private keys on and off devices
+is a security risk.
+
+Then you will need to copy each public certificate to every other
+machine and configure your software so that it trusts that certificate.
+The system will need to repeat this process any time the system adds a
+piece of software. If a machine is compromised, the analogous public
+certificate will need to be untrusted on each node. To combat this
+problem, Certificate Authorities (CA) can help!
+
+#### CAs & Adding Complexity
+
+A CA enables trust deferral from multiple individual certificates to a
+single certificate. Meaning that instead of trusting each certificate,
+each piece of software will trust the CA. The CA will be used to sign
+every public certificate our software pieces need to use. How does
+"signing" work? We will cover that in parts three and four, read on.
+
+Here are the high-level steps:
+
+1. create a CA configuration via OpenSSL CNF files
+2. create the CA
+3. using the CA's public key to sign all of the public certificates
+4. distribute the CA's certificate to every machine
+5. configure the machines certificate store or configure the software
+
+For items one and two, the process can be a bit mystical. There is a
+slew of options involved in managing a CA. To perform number three, you
+will need to go through the processing of creating CSRs on behalf of the
+piece of software, and someone will have to play the role of the CA and
```

Comment:
"and someone will have to play the role of the CA"? someone implies a human - is that intentional?